autoload/ale/handlers/xo.vim

```diff
@@ -0,0 +1,35 @@
+call ale#Set('javascript_xo_executable', 'xo')
+call ale#Set('javascript_xo_use_global', get(g:, 'ale_use_global_executables', 0))
+call ale#Set('javascript_xo_options', '')
+
+call ale#Set('typescript_xo_executable', 'xo')
+call ale#Set('typescript_xo_use_global', get(g:, 'ale_use_global_executables', 0))
+call ale#Set('typescript_xo_options', '')
+
+function! ale#handlers#xo#GetExecutable(buffer, type) abort
+    return ale#node#FindExecutable(a:buffer, a:type . '_xo', [
+    \   'node_modules/xo/cli.js',
+    \   'node_modules/.bin/xo',
+    \])
+endfunction
+
+function! ale#handlers#xo#GetLintCommand(buffer, type) abort
+    return ale#Escape(ale#handlers#xo#GetExecutable(a:buffer, a:type))
+    \   . ale#Pad(ale#handlers#xo#GetOptions(a:buffer, a:type))
+    \   . ' --reporter json --stdin --stdin-filename %s'
+endfunction
+
+function! ale#handlers#xo#GetOptions(buffer, type) abort
+    return ale#Var(a:buffer, a:type . '_xo_options')
+endfunction
+
+" xo uses eslint and the output format is the same
+function! ale#handlers#xo#HandleJSON(buffer, lines) abort
+    return ale#handlers#eslint#HandleJSON(a:buffer, a:lines)
+endfunction
+
+function! ale#handlers#xo#GetProjectRoot(buffer) abort
+    let l:package_path = ale#path#FindNearestFile(a:buffer, 'package.json')
+
+    return empty(l:package_path) ? '' : fnamemodify(l:package_path, ':p:h')
+endfunction
```

Comment:
Admittedly I haven't tested this much in a [monorepo](https://classic.yarnpkg.com/blog/2017/08/02/introducing-workspaces/) setup, so I expect that this might not be exactly what is needed but it [might still be fine](https://github.com/xojs/xo#using-a-parents-config).

Reply:
Changes this to search for `node_modules` the same way `handlers/eslint.vim` does:

https://github.com/dense-analysis/ale/blob/834d6f9c48ce349a4c3d72e98bd0de81aaa84042/autoload/ale/handlers/eslint.vim#L45-L53

Reply:
Ended up walking this back because it isn't necessary. AFAICT [`cosmiconfig`](https://github.com/davidtheclark/cosmiconfig) (which `xo` is using) is already doing this.

Reply:
:+1: